lib/routes/openai/common.ts

```diff
@@ -1,8 +1,76 @@
 import cache from '@/utils/cache';
 import got from '@/utils/got';
+import ofetch from '@/utils/ofetch';
+import { DataItem } from '@/types';
 import { load } from 'cheerio';
 import { art } from '@/utils/render';
 import path from 'node:path';
+import _randUserAgent from '@/utils/rand-user-agent';
+
+export const BASE_URL = new URL('https://openai.com');
+
+/** Fetch the details of an article. */
+export const fetchArticleDetails = async (url: string) => {
+    const page = await ofetch(url);
+    const $ = load(page);
+
+    const $article = $('#main article');
+
+    const categories = $('h1')
+        .prev()
+        .find('a[href]')
+        .toArray()
+        .map((element) => $(element).text());
+
+    // Article header (title, sub title and categories)
+    $($article.find('h1').parents().get(4)).remove();
+    // Related articles (can be the #citations section in some cases, so the last child needs to be removed first)
+    $article.children().last().remove();
+    // Article authors and tags
+    $article.find('#citations').remove();
+
+    return {
+        content: $article.html() ?? undefined,
+        // Categories can be found on https://openai.com/news/ and https://openai.com/research/index/
+        categories,
+        image: $('meta[property="og:image"]').attr('content'),
+    };
+};
+
+/** Fetch all articles from OpenAI's RSS feed. */
+export const fetchArticles = async (limit: number): Promise<DataItem[]> => {
+    const page = await ofetch('https://openai.com/news/rss.xml', {
+        responseType: 'text',
+        headers: {
+            'User-Agent': _randUserAgent({ browser: 'chrome', os: 'windows', device: 'desktop' }),
+        },
```

Comment:
The default UA of chrome on mac https://github.com/DIYgod/RSSHub/blob/b98b4977ff196efbc1d27b8cf771d5c1e407f935/lib/config.ts#L457 works fine